src/Microsoft.PowerShell.ConsoleGuiTools/ConsoleGui.cs

```diff
@@ -56,9 +56,55 @@ public void Start(ApplicationData applicationData)
             top.Add(menu);
 
             var gridHeaders = applicationData.DataTable.DataColumns.Select((c) => c.Label).ToList();
-            // We add one as the offset here to line it up with the data.
-            win.Add(new Label(GetPaddedString(gridHeaders, top.Frame.Width - 3, offset: 1)));
-            
+            var columnWidths = new int[gridHeaders.Count];
+            for (int i = 0; i < gridHeaders.Count; i++)
+            {
+                columnWidths[i] = gridHeaders[i].Length;
+            }
+
+            // calculate the width of each column based on longest string in each column for each row
+            foreach (var row in applicationData.DataTable.Data)
+            {
+                int index = 0;
+                // use just the first 20 rows to calculate in case there is a large dataset
+                foreach (var col in row.Values.Take(20))
```

Comment:
This could be a constant at the top of the file with the comment... Also 20 is fine but maybe we wanna pick like how ever many lines are normally the height of a terminal

Reply:
I'll add it and will use up to half the height of the console as more than that seems unnecessary.